pep-perennial.rst

```diff
@@ -0,0 +1,161 @@
+of pip, so distributors are concerned about moving to a new tag too quickly.
+
+This PEP aims to remove the need for steps 1 and 5 above, so new manylinux tags
+can be adopted more easily.
+
+Naming
+======
+
+Tags using the new scheme will look like::
+
+    manylinux_glibc_2_17_x86_64
+
+Where ``2_17`` is the major and minor version of glibc. I.e. for this example,
+the platform must have glibc 2.17 or newer. Installer tools should be prepared
+to handle any numeric values here, but building and publishing wheels to PyPI
+will probably be constrained to specific profiles defined by auditwheel.
+
+The existing manylinux tags can also be represented in the new scheme:
+
+- ``manylinux1_x86_64`` becomes ``manylinux_glibc_2_5_x86_64``
+- ``manylinux2010_x86_64`` becomes ``manylinux_glibc_2_12_x86_64``
+
+``x86_64`` refers to the CPU architecture, as in previous tags.
+
+While this PEP does not attempt to define tags for non-glibc Linux, the name
+glibc is included to leave room for future efforts in that direction.
+
+Wheel compatibility
+===================
+
+There are two components to a tag definition: a specification of what makes a
+compatible wheel, and of what makes a compatible platform.
+
+A wheel may never use symbols from a newer version of glibc than that indicated
+by its tag. Likewise, a wheel with a glibc tag under this scheme may not be
+linked against another libc implementation.
+
+As with the previous manylinux tags, wheels will be allowed to link against
+a limited set of external libraries and symbols. These will be defined by
+profiles in auditwheel. At least initially, they will likely be similar to
+the list for manylinux2010 (:pep:`571`), and based on library versions in
+newer versions of CentOS.
+
+As with the previous manylinux tags, required libraries which are not on
+the whitelist will need to be bundled into the wheel.
+
+Building compatible wheels
+--------------------------
+
+For each profile defined in auditwheel, there should be a canonical build
+environment, such as a Docker image, available for people to build wheels
+for that profile. People can build in other environments, so long as the
+resulting wheels can be verified by auditwheel, but the canonical environments
+hopefully provide an easy answer for most packages.
+
+The definition of a new profile may well precede the construction of its
+build environment; it's not expected that the definition in auditwheel
+is held up until a corresponding environment is ready to use.
+
+Verification on upload
+----------------------
+
+It is proposed that PyPI will validate manylinux wheels on upload using
+auditwheel, and reject non-compliant packages. This means that only tags for
+which there is a defined profile can be distributed publicly. However,
+organisations may choose to use other tags with this pattern internally,
+for instance if they want to build wheels on Debian instead of CentOS.
```

Comment:
I'm confused about the Debian versus CentOS thing... it could be read as saying that if you want to make a wheel that *only* works on Debian, then it's fine to slap a `manylinux` tag on it internally. Which... I mean, we can't really tell people what to do internally, but using `manylinux` for a wheel that only supports one Linux seems in poor taste, and probably isn't something we want to encourage. (And the plain `linux` tag is already available for folks who want an internal tag with vaguely-defined compatibility that pip will recognize.)

Maybe something like:

> If you need manylinux support for a platform that currently has no profile in auditwheel, then you're encouraged to contribute a profile to auditwheel. If that's not possible for some reason, then other tools can be used, as long as you try to meet the same goal as auditwheel (i.e., the wheel should work in all environments with the given glibc version and architecture) – though you may not be able to upload these wheels to PyPI.